### DISCUSSION TITLE
How to have sub-circuit clock not being an input?

### DISCUSSION THREAD

**Discussion by MarcinOrlowski** [hneemann/Digital#769] [General]
Hi, I got a circuit that drives four 7-segment LED displays. It uses multiplexing so I have additional clock (say plex-clock) to cycle thru all displays. But when I want to use circuit as sub-circuit, `Digital` forces me to provide plex-clock from outside which I do not like really (and yes, I did notice it is called `clock input`). My question: is there any way to have plex-clock hidden?

**Comment by hneemann** [comment]
This is not possible. A clock input is always a simple input if the circuit is used as a sub circuit.
However, usually only one clock input is used and other clock frequencies are then derived from it. If different clock sources were used, the clock signals would drift, causing all sorts of problems.
